### PR TITLE
Passes the current env to the Gutenframe iframe

### PR DESCRIPTION
**Pull request by getdave**
#### Changes proposed in this Pull Request

* Passes the current Calypso env to the Gutenframe `iframe` - this can then be used to selectively enable features on WPCom

#### Questions

* Are there any security implications here?
* This parameter could be manually amended via devtools. Is this considered a problem?
* I had to `--no-verify` this commit because of BEM notices around the use of classes on the `iframe`. Is this ok or not? I'm loath to tamper with these classes unless @gwwar feels otherwise.

#### Testing instructions

* Load Calypso
* Create new Page
* Inspect `iframe` 
* Verify that `environment_id` query arg is on the`iframe` and that it matches your env (likely `development`)

### CODE REVIEW

**Comment by getdave** [comment]
Thanks @marekhrabe. Waiting on 👍 from @gwwar and I'll merge.

File: client/gutenberg/editor/calypsoify-iframe.tsx [hunk 1]
```diff
@@ -490,6 +491,7 @@ const mapStateToProps = ( state, { postId, postType, duplicatePostId }: Props )
 		'frame-nonce': getSiteOption( state, siteId, siteOption ) || '',
 		'jetpack-copy': duplicatePostId,
 		origin: window.location.origin,
+		environment_id: config( 'env_id' ),
```
Comment (gwwar) [inline client/gutenberg/editor/calypsoify-iframe.tsx]: Non-blocking but JS normally uses camelCase vs snake_case

**Comment by gwwar** [comment]
Code here looks okay, what line is lint failing on? We can maybe simply add `/* eslint-disable wpcalypso/jsx-classname-namespace */` to the top of the file

File: client/gutenberg/editor/calypsoify-iframe.tsx (hunk 1 above)
Comment (obenland) [reply client/gutenberg/editor/calypsoify-iframe.tsx]: Is this one of those rare cases where JS caters to PHP, rather than the other way around?

**Comment by getdave** [comment]
> We can maybe simply add /* eslint-disable wpcalypso/jsx-classname-namespace */ to the top of the file

I added this to the specific line. Waiting to see if it's now fixed.

File: client/gutenberg/editor/calypsoify-iframe.tsx (hunk 1 above)
Comment (getdave) [reply client/gutenberg/editor/calypsoify-iframe.tsx]: I've updated to `environment-id ` (with `-`) as that follows the patterns established with `frame-nonce` .etc.

**Pull request merged**